Release 0.5

mmagician opened the pull request:
- fully deprecate circom1, and thus remove make `circom-2` feature the default,
- simplify the traits, we no longer need `CricomBase` vs. `Circom1` & `Circom2`. Directly implement all the methods on the `Wasm` struct,
- update the dependencies to 0.5.0
- adapt version numbering for org consistency

mmagician committed:
move to 0.5 alpha release

mmagician committed:
update mycircuit circom file and its r1cs output

mmagician committed:
gitignore ptau files

mmagician committed:
remove the test case as we no longer have circom to reproduce it

remove unused file

remove further smt stuff

mmagician committed:
remove n_vars, no longer used as its implicit

mmagician committed:
fix groth16 tests

mmagician committed:
deprecate circom1

cleanup dead code

mmagician committed:
update safe circuit (circuit2) code

fix safe circuit

mmagician committed:
provide full paths to wasm location

mmagician committed:
edit msg

mmagician commented on src/witness/witness_calculator.rs:
```diff
@@ -177,72 +115,9 @@ impl WitnessCalculator {
-            .set_free_pos(store, old_mem_free_pos)?;
-
-        Ok(w)
+        self.calculate_witness_circom2(store, inputs)
```
note to self, this should be inlined

mmagician committed:
remove unused files

mmagician committed:
removed unused `version` key from `rust-toolchain.toml`, use latest stable

Co-authored-by: jakehemmerle <jakehemmerle@protonmail.com>

mmagician committed:
use stable rust in CI

mmagician committed:
inline code

mmagician committed:
cargo update to catch the latest groth16 release

mmagician commented:
This code works now after https://github.com/arkworks-rs/groth16/pull/61 has been merged and groth 0.5.0-alpha.1 released. Let's keep it as a branch for now till we upgrade all other repos

mmagician committed:
remove deprecated feature flag from CI

anudit commented:
@mmagician Just tried the `0.5.0-alpha.1` release, proof verification seems to work 🚀 

![image](https://github.com/user-attachments/assets/e90a8baa-e5dd-42e7-8d25-42c344ed14d1)

mmagician committed:
Rust limitation no longer applies; warning about circom/arkworks diff

mmagician committed:
release numbering consistency across arkworks

mmagician committed:
add homepage & repo manifest fields

Pratyush commented:
Thanks @mmagician; could you also update the README to change the line about adding `ark-circom` as a dependency to use the crates version instead?

Also, @tyshko-rostyslav, could you please authorize the crates-io team within the arkworks organization to make releases for the `ark-circom` crate?

mmagician committed:
Merge branch 'master' into release-0.5

CPerezz committed:
fix: Wasm compilation for wasm32-wasi target (#74)

After forcing `wasmer-wasix` to use the `js` feature when the
`circom-compat/js` feature is enabled. As well as forcing `wasmer/js`
and otherwise, `wasmer/default`. The crate can now successfully compile
to wasm.

Resolves: #73

tyshko-rostyslav commented:
> Thanks @mmagician; could you also update the README to change the line about adding `ark-circom` as a dependency to use the crates version instead?
> 
> Also, @tyshko-rostyslav, could you please authorize the crates-io team within the arkworks organization to make releases for the `ark-circom` crate?

Sure, but as far as I understand from [docs](https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner) you need to have a team in arkworks-rs created. Otherwise I could add people one by one, whatever is easier for you

tyshko-rostyslav commented:
> @tyshko-rostyslav There is a `crates.io` team in the arkworks-rs organization: https://crates.io/teams/github:arkworks-rs:crates-io

As per [docs](https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner) I have to be a member of your team to be able to give the team access. Could you please (temporarily) add me to that team?

mmagician commented:
@Pratyush

tyshko-rostyslav commented:
> @Pratyush

Thanks! Added the team

mmagician committed:
ignore cargo.lock

mmagician committed:
update wasix version

mmagician committed:
add a wasm compilation CI check

mmagician commented:
Thanks @tyshko-rostyslav! https://crates.io/crates/ark-circom/0.5.0-alpha

mmagician commented:
@CPerezz I ended up upgrading the dependencies for `wasmer`. I added CI for this as well, but could you still confirm the new alpha release works on your setup please?

CPerezz commented:
> @CPerezz I ended up upgrading the dependencies for `wasmer`. I added CI for this as well, but could you still confirm the new alpha release works on your setup please?

Hey, this target and setup work for the `build` and `check`. Also, couple things to at least mention.
- Target `wasm32-wasi` also works. So maybe worth to test for a `matrix{{}}` of targets instead in the CI?
- I've found sometimes that `check` or `build` won't compile the lib as if it was a dependency. Resulting sometimes in unexpected errors when used downstream. 
I usually have this script running in the CI to asses that indeed no errors happen even when the lib is a dependency of another crate: https://github.com/privacy-scaling-explorations/sonobe/blob/main/.github/scripts/wasm-target-test-build.sh
And I call it like: https://github.com/privacy-scaling-explorations/sonobe/blob/0ad54576ec6462d63313ebde18bb0a24669fd00e/.github/workflows/ci.yml#L104


In any case, definitely looks well! Nice work!

mmagician committed:
Merge branch 'master' into release-0.5

mmagician committed:
bump versions

mmagician committed:
update crate reference in the README

AlaaZorkane commented:
any updates on this 🙏?
